web3.nim

```diff
@@ -700,17 +741,28 @@ proc send*(c: ContractCallBase, value = 0.u256, gas = 3000000'u64, gasPrice = 0)
     cc.gasPrice = some(gasPrice)
   web3.send(cc)
 
-proc call*[T](c: ContractCall[T], value = 0.u256, gas = 3000000'u64): Future[T] {.async.} =
+proc call*[T](c: ContractCall[T],
+              value = 0.u256,
+              gas = 3000000'u64,
+              blockNumber = high(uint64)): Future[T] {.async.} =
   var cc: EthCall
   cc.data = some("0x" & c.data)
   cc.source = some(c.web3.defaultAccount)
   cc.to = c.to
   cc.gas = some(Quantity(gas))
   cc.value = some(value)
-  let response = await c.web3.provider.eth_call(cc, "latest")
-  var res: T
-  discard decode(strip0xPrefix(response), 0, res)
-  return res
+  let response = strip0xPrefix:
+    if blockNumber != high(uint64):
+      await c.web3.provider.eth_call(cc, blockNumber)
```

Comment:
out of curiosity, it should be possible to do `await if: ... else: ...` here, no?